src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenIfOperator.vb

```diff
@@ -1247,5 +1247,105 @@ Friend Module BIFOpResult0011mod
 }]]>)
         End Sub
 
+        <Fact>
+        Public Sub IfOnUnconstrainedMethodTypeParameter()
+            CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Friend Module Mod1
+        Sub M1(Of T)(arg1 As T, arg2 As T)
+            System.Console.WriteLine(If(arg1, arg2))
+        End Sub
+
+        Sub Main()
+            M1(Nothing, 10)
+            M1(1, 10)
+            M1(Nothing, "String Parameter 1")
+            M1("String Parameter 2", "Should not print")
+            M1(Of Integer?)(Nothing, 4)
+            M1(Of Integer?)(5, 10)
+        End Sub
+    End Module
+    </file>
+</compilation>, expectedOutput:=<![CDATA[
+0
+1
+String Parameter 1
+String Parameter 2
+4
+5
+]]>).VerifyIL("Mod1.M1", <![CDATA[
+{
+  // Code size       22 (0x16)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  box        "T"
+  IL_0007:  brtrue.s   IL_000b
+  IL_0009:  pop
+  IL_000a:  ldarg.1
+  IL_000b:  box        "T"
+  IL_0010:  call       "Sub System.Console.WriteLine(Object)"
+  IL_0015:  ret
+}
+]]>)
+        End Sub
+
+        <Fact>
+        Public Sub IfOnUnconstrainedTypeParameterWithNothingLHS()
+            CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Friend Module Mod1
+        Sub M1(Of T)(arg As T)
+            Console.WriteLine(If(Nothing, arg))
+        End Sub
+
+        Sub Main()
+            ' Note that this behavior is different than C#'s behavior. This is consistent with Roslyn's handling
+            ' of If(Nothing, 1), which will evaluate to 1
+            M1(1)
+            Console.WriteLine(If(Nothing, 1))
+            M1("String Parameter 1")
+            M1(Of Integer?)(3)
+        End Sub
+    End Module
+    </file>
+</compilation>, expectedOutput:=<![CDATA[
+1
+1
+String Parameter 1
+3
+]]>).VerifyIL("Mod1.M1", <![CDATA[
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  .locals init (T V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  box        "T"
+  IL_0006:  call       "Sub System.Console.WriteLine(Object)"
+  IL_000b:  ret
+}
+]]>)
+        End Sub
+
+        <Fact>
+        Public Sub IfOnUnconstrainedTypeParametersOldLanguageVersion()
+            CreateCompilation(
+<compilation>
+    <file name="a.vb">
+Friend Module Mod1
+        Sub M1(Of T)(arg1 As T, arg2 As T)
+            System.Console.WriteLine(If(arg1, arg2))
+        End Sub
+    End Module
+    </file>
+</compilation>, parseOptions:=TestOptions.Regular15_5).AssertTheseDiagnostics(<![CDATA[
+BC36716: Visual Basic 15.5 does not support unconstrained type parameter variables in two operand conditionals.
```

Comment:
No information which version does?

Reply:
BC36716 doesn't have this information for any feature that uses it, going back to VB 9.0. We can add it if we want to, but that is out of scope for this PR.

---
In reply to: [217216426](https://github.com/dotnet/roslyn/pull/29724#discussion_r217216426) [](ancestors = 217216426)

Reply:
> We can add it if we want to, but that is out of scope for this PR.

Could you open an issue to follow up?

---
In reply to: [217221037](https://github.com/dotnet/roslyn/pull/29724#discussion_r217221037) [](ancestors = 217221037,217216426)

Reply:
Created [https://github.com/dotnet/roslyn/issues/29874](https://github.com/dotnet/roslyn/issues/29874)

---
In reply to: [217559319](https://github.com/dotnet/roslyn/pull/29724#discussion_r217559319) [](ancestors = 217559319,217221037,217216426)